app/main.cpp

```diff
@@ -299,8 +302,31 @@ int main(int argc, char *argv[])
 
     QQuickStyle::setStyle("Material");
 
-    // Load the main.qml file
     QQmlApplicationEngine engine;
+    QString initialView;
+
+    GlobalCommandLineParser parser;
+    switch (parser.parse(app.arguments())) {
+    case GlobalCommandLineParser::NormalStartRequested:
+        initialView = "PcView.qml";
+        break;
+    case GlobalCommandLineParser::StreamRequested:
+        {
+            initialView = "CliStartStreamSegue.qml";
+            StreamingPreferences* preferences = new StreamingPreferences(&app);
```

Comment:
This StreamingPreferences object needs to be deleted at some point.

Reply:
Note that I'm passing the `QCoreApplication` as a parent to the `StreamingPreferences`. That handles deletion of the preferences object automatically when `QCoreApplication` object gets deleted.